client/src/app/shared/angular/timestamp-route-transformer.directive.ts

```diff
@@ -0,0 +1,34 @@
+import { Directive, ElementRef, HostListener, Output, EventEmitter } from '@angular/core'
+import { Router } from '@angular/router'
+
+type ElementEvent = Omit<Event, 'target'> & {
+  target: HTMLInputElement
+}
+
+@Directive({
+  selector: '[timestampRouteTransformer]'
+})
+export class TimestampRouteTransformerDirective {
+  @Output() timestampClicked = new EventEmitter<number>()
+
+  constructor (private el: ElementRef, private router: Router) { }
+
+  @HostListener('click', ['$event'])
+  public onClick ($event: ElementEvent) {
+    if ($event.target.hasAttribute('href')) {
+      const ngxLink = document.createElement('a')
+      ngxLink.href = $event.target.getAttribute('href')
+
+      // we only care about reflective links
+      if (ngxLink.host !== window.location.host) return
+
+      const ngxLinkParams = new URLSearchParams(ngxLink.search)
+      if (ngxLinkParams.has('start')) this.timestampClicked.emit(parseInt(ngxLinkParams.get('start'), 10))
+
+      this.router.navigateByUrl(ngxLink.pathname + ngxLink.search + ngxLink.hash)
```

Comment:
I'm not sure we should update the link (youtube does not seem to update it), because you could have some undesired effects: you handle URL param change + handle the link click in the watch page

Reply:
I can remove the link update altogether.

Reply:
I think keeping the link is fine (in case where your open the link in another tab)